src/server/src/main/java/io/cassandrareaper/service/RepairRunService.java

```diff
@@ -177,18 +175,12 @@ public RepairRun registerRepairRun(
 
   static int computeGlobalSegmentCount(
       int segmentCountPerNode,
-      Map<List<String>, List<String>> rangeToEndpoint,
       Map<String, List<RingRange>> endpointToRange) {
 
     int nodeCount = Math.max(1, endpointToRange.keySet().size());
```

Comment:
is `endpointToRange.keySet().size()` ever less than one?

_if it were me_ i would tighten things up here.
`rangeToEndpoint` comes from `buildEndpointToRangeMap(..)` which in turn comes from `jmxProxy.getRangeToEndpointMap(..)`.

If there are no endpoints for the range in question we've got ourselves a programmatic error and we should catch that accordingly.

So the method could be written like:
```
  static int computeGlobalSegmentCount(int segmentCountPerNode, Map<String, List<RingRange>> endpointToRange) {
    Preconditions.checkArgument(1 <= endpointToRange.keySet().size());

    return endpointToRange.keySet().size()
        * (segmentCountPerNode != 0 ? segmentCountPerNode : DEFAULT_SEGMENT_COUNT_PER_NODE);
  }
```
(i'd also put an assert check in buildEndpointToRangeMap, before the return statement)